src/working/quick_sort.py

```diff
@@ -0,0 +1,75 @@
+"""Quick sort data structure."""
+
+
+def quicksort(data):
+    """Sorting algorithm for quick sort."""
+    try:
+        if len(data) > 1:
+            mid = data[0]
+            left = []
+            right = []
+
+            for item in data[1:]:
+                # This slows it down by almost a second but covers 2.7
+                if not isinstance(item, type(mid)):
+                    raise TypeError
```

Comment:
If you raise a TypeError, use `raise TypeError('some text')` to help with error handling